exhibitor-core/src/main/java/com/netflix/exhibitor/core/config/azure/AzureConfigProvider.java

```diff
@@ -0,0 +1,147 @@
+     * @param arguments   args
+     * @param hostname    this VM's hostname
+     * @throws Exception errors
+     */
+    public AzureConfigProvider(AzureClientFactory factory, AzureCredential credentials, AzureConfigArguments arguments, String hostname) throws Exception {
+        this(factory, credentials, arguments, hostname, new Properties());
+    }
+
+    /**
+     * @param factory     the factory
+     * @param credentials credentials
+     * @param arguments   args
+     * @param hostname    this VM's hostname
+     * @param defaults    default props
+     * @throws Exception errors
+     */
+    public AzureConfigProvider(AzureClientFactory factory, AzureCredential credentials, AzureConfigArguments arguments, String hostname, Properties defaults) throws Exception {
+        this.arguments = arguments;
+        this.hostname = hostname;
+        this.defaults = defaults;
+        azureClient = factory.makeNewClient(credentials);
+    }
+
+    public AzureClient getAzureClient() {
+        return azureClient;
+    }
+
+    @Override
+    public void start() throws Exception {
+        // NOP
+    }
+
+    @Override
+    public void close() throws IOException {
+        azureClient.close();
+    }
+
+    @Override
+    public PseudoLock newPseudoLock() throws Exception {
+        return new AzurePseudoLock
+                (
+                        azureClient,
+                        arguments.getContainer(),
+                        arguments.getLockArguments().getPrefix(),
+                        arguments.getLockArguments().getTimeoutMs(),
+                        arguments.getLockArguments().getPollingMs(),
+                        arguments.getLockArguments().getSettlingMs()
+                );
+    }
+
+    @Override
+    public LoadedInstanceConfig loadConfig() throws Exception {
+        Date lastModified;
+        Properties properties = new Properties();
+        CloudBlob object = getConfigObject();
+        if (object != null) {
+            lastModified = object.getProperties().getLastModified();
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            object.download(out);
+            properties.load(new ByteArrayInputStream(out.toByteArray()));
+        } else {
+            lastModified = new Date(0L);
+        }
+
+        PropertyBasedInstanceConfig config = new PropertyBasedInstanceConfig(properties, defaults);
+        return new LoadedInstanceConfig(config, lastModified.getTime());
+    }
+
+    @Override
+    public LoadedInstanceConfig storeConfig(ConfigCollection config, long compareVersion) throws Exception {
+        {
+            BlobProperties blobProperties = getConfigBlobProperties();
+            if (blobProperties != null) {
+                Date lastModified = blobProperties.getLastModified();
+                if (lastModified.getTime() != compareVersion) {
+                    return null;    // pattern copied from S3ConfigProvider - Azure may support a better way
```

Comment:
please add this as a TODO:
    // TODO(marco): blah blah
and avoid long lines (the // comment should be on the line above the offending code)